pkg/docker/volume.go

```diff
@@ -30,16 +31,15 @@ func EnsureVolume(ctx context.Context, log *zap.SugaredLogger, cli *client.Clien
 	log.Debug("checking state of volume")
 
 	// Check whether volume exists.
-	volumes, err := cli.VolumeList(ctx, filters.Args{})
+	exactName := fmt.Sprintf("^%s$", opts.Name)
+	volumes, err := cli.VolumeList(ctx, filters.NewArgs(filters.Arg("name", exactName)))
 	if err != nil {
 		return nil, false, err
 	}
 
-	for _, v := range volumes.Volumes {
-		if v.Name == opts.Name { // We found a match, volume exists.
-			log.Info("found existing volume")
-			return v, false, nil
-		}
+	if len(volumes.Volumes) > 0 {
+		log.Info("found existing volume")
+		return volumes.Volumes[0], false, err
```

Comment:
Logically speaking, this would be sound if our length check was for == 1. But in the current form, it's unsound, because we allow for N volumes to be returned, yet assume the first one is the correct one. We should either:

1. add an assertion check that len == 1 and error if it's higher, or
2. be lenient as before.

It's a good practice to be defensive when dealing with third party libraries, especially Docker... 🙃